### PR TITLE
Fix longformer parity and perf regression

### PR DESCRIPTION
**Pull request by tianleiwu**
**Description**: 

(1) Restore original CUDA kernel as default, and add an environment variable to enable new CUDA kernel when memory is a concern.
(2) Update convert_longformer_to_onnx: support  transformers 4.3.2, where an extra parameter is added to LongformerSelfAttention.forward; add an option --export_padding so that user could choose whether to have padding logic in ONNX model.
(3) Improve benchmark_longformer tool: add memory measurement

To use new kernel, set environment variable ORT_LONGFORMER_COMPACT_MEMORY=1. Note that when sequence_length <= 2 * attention_window or global_tokens > attention_window, we will use fast kernel and ignore the environment variable. This is for experimental purpose, and the environment variable might be removed in the future.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

Walk around a few issues in https://github.com/microsoft/onnxruntime/pull/6646:
(1) The result is different with huggingface transformers when sequence_length == 2 * attention_window.
(2) Kernel using compact memory is slower
(3) There is additional constraint that number of global tokens <= attention_window.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/longformer/convert_longformer_to_onnx.py [hunk 1]
```diff
@@ -80,134 +78,67 @@ def parse_arguments():
...
-    assert attn_output.size() == hidden_states.size(), "Unexpected size"
+    # TODO: add postprocess of ONNX model to calculate based on graph input: input_mask = (attention_mask - 1) * 10000.0
+    # TODO: add postprocess of ONNX model to use graph input directly: glboal_mask = global_attention_mask
 
-    outputs = (attn_output, )
-    return outputs
-
-
-# For transformers 3.0
-def my_longformer_attention_forward_3(self, hidden_states, attention_mask, output_attentions=False):
```
Comment (wangyems) [inline onnxruntime/python/tools/transformers/longformer/convert_longformer_to_onnx.py:186]: Will we support transformers 3.0?

Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/longformer/convert_longformer_to_onnx.py:186]: I removed the support for transformers 3.0 since the ONNX graph for 3.0 is slower. In this way, we could push user to use 4.0 for better performance, and reduce maintenance cost.

**Review by wangyems — DISMISSED** [review]

**Review by wangyems — APPROVED** [review]
